python_ci.yml

```diff
@@ -1,26 +1,52 @@
 version: 2.1
 
 jobs:
+    install_py3:
+        docker:
+            - image: svalinn/pymoab-py3-18.04
+        steps:
+            - checkout #pull repo in the docker container
+            - run: pip install --user
+            - run: 
+              working_directory: ~/
+              command: python -c "import my_package"
+                
+    install_py2:
+        docker:
+            - image: svalinn/pymoab-py2-18.04
+        steps:
+            - checkout #pull repo in the docker container
+            - run: pip install --user
+            - run: 
+              working_directory: ~/
+              command: python -c "import my_package"
+    
     run_test_py3:
         docker:
             - image: svalinn/pymoab-py3-18.04
         steps:
             - checkout #pull repo in the docker container
             - run:
-                working_drectory: #/directory
-            command: python -m pytest
+                working_directory: #/directory
+                command: python -m pytest
 
     run_test_py2:
         docker:
             - image: svalinn/pymoab-py2-18.04
         steps:
             - checkout #pull repo in the docker container
             - run:
-                working_drectory: #/directory
-            command: python -m pytest
+                working_directory: #/directory
+                command: python -m pytest
 
 workflows:
     my_workflows:
         jobs:
-            - run_test_py3
-            - run_test_py2
+            - install_py3
+            - run_test_py3:
+                - requires:
+                    install_py3
```

Comment:
I strongly disagree about this dependency. If anything the dependency should be the other way around.

Fundamental testing should be against the version that is in the repo and NOT against the version that is installed.  CI is testing the PR to the repo.  If something goes wrong (without failing) in the installation, you could get false positives or negatives.

Arguably, there should be no attempt to install until we are sure the tests have passed on the code in the repo.

Reply:
I completely agree with your statement!

I guess I Think about the standard way to do things... install and then test but in this case it does not make sense!

Will fix this tomorrow